### PR TITLE
Add flow_graph accumulate method

### PR DESCRIPTION
**Pull request by adriendelsalle**
Description
--

Add `flow_graph` `accumulate` method to be able to accumulate a data upon the flow graph.
The accumulator is limited to a hard-coded sum functor.

- [ ] Rebase before merge

### CODE REVIEW

File: include/fastscapelib/structured_grid.hpp [hunk 1]
```diff
@@ -330,6 +330,8 @@ namespace fastscapelib
 
         const node_status_type& status_at_nodes() const;
 
+        distance_type node_area(const size_type& idx) const noexcept;
```
Comment (benbovy) [inline include/fastscapelib/structured_grid.hpp:333]:
```suggestion
        distance_type cell_area(const size_type& idx) const noexcept;
```

Could we also create a `area_type` alias? It's another dumb alias, but `distance_type` sounds very odd for a cell area.

**Review by benbovy — COMMENTED** [review]
Thanks @adriendelsalle! I've left one comment below.

File: include/fastscapelib/structured_grid.hpp (hunk 1 above)
Comment (adriendelsalle) [reply include/fastscapelib/structured_grid.hpp:333]: Sure :)

Comment (adriendelsalle) [reply include/fastscapelib/structured_grid.hpp:333]: Let's improve this in a new PR if it's OK for you

Comment (benbovy) [reply include/fastscapelib/structured_grid.hpp:333]: Yes sure.

**Pull request merged**